src/model/model.ts

```diff
@@ -220,7 +220,19 @@ export class Model {
       if (!ast) {
         return false;
       }
-      ast.policy.push(rule);
+
+      const policy = ast.policy;
+      const tokens = ast.tokens;
+
+      if (tokens.find((str) => str === 'p_priority')) {
```

Comment:
We must consider the performance. You can do like so: 
1. find `p_priority ` index from tokens.
2. if `p_priority ` index is not -1, find the insertion index from the policy and insert the rule after this index, otherwise push the rule to the policy.

Reply:
In fact I'd like to implement like this
> When the effector judge the effect, set a flag on the model

Reply:
It should be more complicated in effector.